### ISSUE TITLE
Are UP GT-s correct when training with large distortion?

### ISSUE THREAD

**Issue by suvigy** [cvg/GeoCalib#24]
Hi!

 I'm trying to train with simple Division camera model, which can also model wide angle fisheye camera models. I created the training data set with the following params, I just hightlight the relevant parts.

```
....
camera_model: simple_divisional
parameter_dists:
  vfov:
     ...
       loc: 1.88
       scale: 0.01 
    ....
  k1:
    .....
      loc:  -0.29104758148391235
      scale: 0.002
.....
 shape:
    type: fix
    value:
      - 640
      - 960

```

In wandb I'm getting the following visualization also including the up gt, pred, confidence:

![Image](https://github.com/user-attachments/assets/09b4e8bc-f1f9-4ceb-a240-45552a0fcade)

Do you think the 'Up Gt' is correct? To me, it seems that there is mostly an inner circle within which the up vectors are correct, but outside of this circle, they appear to be incorrect. Or should it be interpreted in a different way?
(As shown in the example, out-of-FoV pixels are not yet masked. The simple_division model sets every pixel as valid during undistortion. I hope the UpGt issue isn't related to this; I plan to address the valid pixels later.)

**Comment by veichta** [comment]
Hi,
Very interesting! Would be happy to know how it goes.

My best bet at this point is that using such a big FoV together with strong distortions will produce invalid pixels in the distortion function [here](https://github.com/cvg/GeoCalib/blob/fbd383d6953959a5b1af3cfc6b7f348bcc667bf8/siclib/geometry/camera.py#L163) since you need ```(1 - 4 * self.k1[..., None, None] * r2) >= 0```. 

Unfortunately, we never trained on data where this became relevant, so the check_valid is not used in the most of the code.

**Comment by suvigy** [comment]
I think it is bug in the dataloader, here:
https://github.com/cvg/GeoCalib/blob/main/siclib/datasets/simple_dataset.py#L149C18-L149C30
it is always the SimpleRadial which is created no matter if the SimpleDivisional cameral specified in the LM optimizer config. 
Currently I just quick-fixed it on my side simply by creating the SimpleDivisional camera model in the dataloader (for the time being hard code, no config check, in the config simple divisional is set for lm optimizer)
